File: bubble-sort/bubble-sort.clj
(defn bubble-sort [arr]
  (loop [arr arr n (count arr)]
    (if (<= n 1)
      arr
      (let [[new-arr swapped] 
            (reduce (fn [[acc swapped] i]
                     (if (and (< (inc i) (count acc))
                             (> (nth acc i) (nth acc (inc i))))
                       [(assoc acc i (nth acc (inc i)) (inc i) (nth acc i)) true]
                       [acc swapped]))
                   [arr false] (range (dec n)))]
        (if swapped
          (recur new-arr (dec n))
          new-arr)))))

(println (bubble-sort [64 34 25 12 22 11 90]))

File: bubble-sort/bubble-sort.cs
using System;

class BubbleSort {
    static void Sort(int[] arr) {
        int n = arr.Length;
        for (int i = 0; i < n - 1; i++) {
            for (int j = 0; j < n - i - 1; j++) {
                if (arr[j] > arr[j + 1]) {
                    (arr[j], arr[j + 1]) = (arr[j + 1], arr[j]);
                }
            }
        }
    }
    
    static void Main() {
        int[] arr = {64, 34, 25, 12, 22, 11, 90};
        Sort(arr);
        Console.WriteLine(string.Join(" ", arr));
    }
}
